### DISCUSSION TITLE
Packaging by layer or by use case? Or by both?

### DISCUSSION THREAD

**Discussion by stephaniesaab** [CSC207-2022F-UofT/course-project-my-dog-ate-my-hw#19] [General]
Hi everyone, we've been having a little trouble deciding on exactly how to package our code.

**Comment by stephaniesaab** [comment]
We're currently deciding on packaging by layer or by use case, or by both. In the case of both, the entities and frameworks/drivers would each be 'shared packages' that no one is solely responsible for, because they're used by everyone. The interactors and input/output boundaries for each case would be specific packages that are under the responsibility of the coder in charge of that particular use case. What do we think about this idea?

**Reply by BetterthanNothin** [reply]
I think the smartest thing to do is to keep the input layers and output layers together, and keep the use case interactors together so that it we can manage things more easily.

**Reply by XinyingXu** [reply]
Then maybe we package them by layers entirely

**Reply by stephaniesaab** [reply]
The only thing with packaging by layer may make it more difficult to assign responsibility for each use case, do our use cases share enough features to be packaged by layer?

**Reply by XinyingXu** [reply]
Like the password manager, they have folders: entities, controller, UI, userCases

**Reply by XinyingXu** [reply]
I think they share enough features.

**Reply by stephaniesaab** [reply]
Looking at the password manager example I think that makes sense, they still have a folder for use cases though. I think it's a good idea to have one folder for the pictures/icons of the pet so that all the png files are somewhere known. For the use cases: where should we put the input/output boundaries? Maybe we can have the 'UI' folder with subfolders for 'input' and 'output' boundaries?

**Reply by XinyingXu** [reply]
I think it makes more sense if we have subfolders for each use case under 'useCases', and pack the input/output/interactors there.

**Reply by stephaniesaab** [reply]
I agree
